### ISSUE TITLE
Gitlab MR decoration failed (Cannot format given Object as a Number)

### ISSUE THREAD

**Issue by Atharex** [mc1arke/sonarqube-community-branch-plugin#325]
**Describe the bug**
The plugin enables importing Gitlab projects and scanning pull requests correctly, but the MR decoration fails.

**To Reproduce**

```
sonar-scanner   -Dsonar.projectKey=gitlab_project_AXinYBuZsjqXafDIc7B1   -Dsonar.sources=.   -Dsonar.host.url=<sonarqube-url>   -Dsonar.login=<token> -Dsonar.pullrequest.key=4 -Dsonar.pullrequest.branch=feature/458 -Dsonar.pullrequest.base=master
```

This correctly analyzes the branch and compares it to master. However sonarqube logs show that decoration has failed ("Cannot format given Object as a Number") with the following output:

```
...
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Webhooks | globalWebhooks=0 | projectWebhooks=0 | status=SUCCESS | time=3ms
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.PullRequestPostAnalysisTask] using pull request decorator com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabServerPullRequestDecorator
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] starting to analyze with com.github.mc1arke.sonarqube.plugin.ce.pullrequest.AnalysisDetails@631b8200
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] class com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.response.MergeRequest received
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] Status url is: https://gitlab.com/api/v4//projects/<project-id>/statuses/3c3bd88b77c09cb46dc97450ae2f52c0b46c5173
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] PR commits url is: https://gitlab.com/api/v4//projects/<project-id>/merge_requests/4/commits
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] MR discussion url is: https://gitlab.com/api/v4//projects/<project-id>/merge_requests/4/discussions
2021.04.06 13:51:37 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] User url is: https://gitlab.com/api/v4//user
2021.04.06 13:51:38 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] class com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.response.User received
2021.04.06 13:51:38 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] Using user: <my-user>
2021.04.06 13:51:38 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] MR discussions received
2021.04.06 13:51:39 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] MR discussions received
2021.04.06 13:51:39 INFO  ce[AXinc-8ksjqXafDIc7CN][c.g.m.s.p.c.p.g.GitlabServerPullRequestDecorator] Discussions in MR: 9b149d081b3f910e14649eedaff5db853bb81a38
2021.04.06 13:51:39 ERROR ce[AXinc-8ksjqXafDIc7CN][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Execution of task class com.github.mc1arke.sonarqube.plugin.ce.pullrequest.PullRequestPostAnalysisTask failed
java.lang.IllegalArgumentException: Cannot format given Object as a Number
        at java.base/java.text.DecimalFormat.format(Unknown Source)
        at java.base/java.text.Format.format(Unknown Source)
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.AnalysisDetails.createAnalysisSummary(AnalysisDetails.java:207)
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabServerPullRequestDecorator.decorateQualityGateStatus(GitlabServerPullRequestDecorator.java:171)
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.PullRequestPostAnalysisTask.finished(PullRequestPostAnalysisTask.java:160)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.executeTask(PostProjectAnalysisTasksExecutor.java:118)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.finished(PostProjectAnalysisTasksExecutor.java:109)
        at org.sonar.ce.task.step.ComputationStepExecutor.executeListener(ComputationStepExecutor.java:91)
        at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:63)
        at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:235)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:217)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:162)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:137)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:89)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
2021.04.06 13:51:39 INFO  ce[AXinc-8ksjqXafDIc7CN][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Pull Request Decoration | status=FAILED | time=1747ms
2021.04.06 13:51:39 INFO  ce[AXinc-8ksjqXafDIc7CN][o.s.c.t.CeWorkerImpl] Executed task | project=gitlab_project_AXinYBuZsjqXafDIc7B1 | type=REPORT | pullRequest=4 | id=AXinc-8ksjqXafDIc7CN | submitter=admin | status=SUCCESS | time=2107ms

```

**Expected behavior**
This failure should not happen and the decoration should work correctly

**Software Versions**
 - SonarQube Version: [e.g. Version 8.7.1 (build 42226)]
 - Plugin Version: 1.7.0

**Comment by rockmagic** [comment]
#335

**Comment by mc1arke** [comment]
Fixed in 1.8.0 of the plugin
